src/System.Buffers.Primitives/System/Buffers/ReadOnlyBufferReader.cs

```diff
@@ -94,36 +98,143 @@ private void MoveNext()
             _end = true;
         }
 
-        public void Skip(int length)
+        public void Skip(int byteCount)
         {
-            if (length < 0)
+            if (byteCount < 0)
             {
                 ThrowHelper.ThrowArgumentOutOfRangeException(ExceptionArgument.length);
             }
 
-            _consumedBytes += length;
+            _consumedBytes += byteCount;
 
-            while (!_end && length > 0)
+            while (!_end && byteCount > 0)
             {
-                if ((_index + length) < _currentSpan.Length)
+                if ((_index + byteCount) < _currentSpan.Length)
                 {
-                    _index += length;
-                    length = 0;
+                    _index += byteCount;
+                    byteCount = 0;
                     break;
                 }
 
                 var remaining = (_currentSpan.Length - _index);
 
                 _index += remaining;
-                length -= remaining;
+                byteCount -= remaining;
 
                 MoveNext();
             }
 
-            if (length > 0)
+            if (byteCount > 0)
             {
                 ThrowHelper.ThrowArgumentOutOfRangeException(ExceptionArgument.length);
             }
         }
+
+        public void SkipTo(Position position)
+        {
+            _currentPosition = position;
+            _nextPosition = position;
+            if (_sequence.TryGet(ref _nextPosition, out ReadOnlyMemory<byte> memory))
+            {
+                _currentSpan = memory.Span;
+            }
+            else
+            {
+                _currentSpan = default;
+            }
+            _index = 0;
+        }
+      
+        public Position? PositionOf(byte value)
+        {
+            var unread = UnreadSegment;
+            var index = unread.IndexOf(value);
+            if (index != -1) return _currentPosition + (_index + index);
+
+            var nextSegmentPosition = _nextPosition;
+            var currentSegmentPosition = _nextPosition;
+            while (_sequence.TryGet(ref nextSegmentPosition, out var memory, true))
+            {
+                var segment = memory.Span;
+                index = segment.IndexOf(value);
+                if (index != -1) return currentSegmentPosition + index;
+                currentSegmentPosition = nextSegmentPosition;
+            }
+            return default;
+        }
+
+        public Position? PositionOf(ReadOnlySpan<byte> value)
```

Comment:
Can we remove this for now? I'd like to keep this API minimal and fully tested. Expose whatever needs to be exposed to implement this in the experimental APIs

Reply:
I don't think it can be removed, and I don't think we can sweep the issue under the rug. The remaining abstraction is not powerful enough to implement reading of variable length data (based on searches). We would have to expose many fields of the reader (in addition to the sequence) to implement for example ReadUntill, at which point using the reader would be very error prone as the callers could mess up its state too easily.